.github/workflows/main.yml

```diff
@@ -25,30 +25,64 @@ jobs:
       matrix:
         java_version: [ 8, 11, 16 ]
         scala_version: [ 2.12.11 ]
-        os: [ 'ubuntu-latest', 'windows-latest' ]
-    env:
-      SBT: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++${{ matrix.scala_version }} coverage
-      SBTNOCOV: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++${{ matrix.scala_version }}
+        os: [ ubuntu-20.04, windows-2019 ]
+        include:
+          - os: ubuntu-20.04
+            shell: bash
+          - os: windows-2019
+            shell: msys2 {0}
 
     runs-on: ${{ matrix.os }}
     defaults:
       run:
-        shell: bash
-
+        shell: ${{ matrix.shell }}
+    env:
+      SBT: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++${{ matrix.scala_version }} coverage
+      SBTNOCOV: sbt -J-Xms1024m -J-Xmx5120m -J-XX:ReservedCodeCacheSize=512m -J-XX:MaxMetaspaceSize=1024m ++${{ matrix.scala_version }}
     steps:
 
       ############################################################
       # Setup
       ############################################################
 
-      - name: Checkout Repository
-        uses: actions/checkout@v2.0.0
+      - name: Install Dependencies (Linux)
+        if: runner.os == 'Linux'
+        run: sudo apt-get install -y libmxml-dev
+
+      - name: Install Dependencies (Windows)
+        if: runner.os == 'Windows'
+        uses: msys2/setup-msys2@v2
+        with:
+          install: gcc libargp-devel make pkgconf
```

Comment:
As to libargb being widely distributed, I dont' think that matters to ASF policy. The [license page](https://www.apache.org/legal/resolved.html) says the following about dependencies to components with restricted licenses:

> Apache projects can rely on components under prohibited licenses if the component is only needed for optional features. When doing so, a project shall provide the user with instructions on how to obtain and install the non-included work. Optional means that the component is not required for standard use of the product or for the product to achieve a desirable level of quality. The question to ask yourself in this situation is:
>
> "Will the majority of users want to use my product without adding the optional components?"

So we just need to answer this question.

From my perspective, if we generate code, I would think most users would expect that code to compile to the daffodil/daffodil.exe binary and then use that binary, even if libargp weren't installed. If a user doesn't have that installed, I suspect right now compilation will fail?

Perhaps if we could split things up so we don't build the CLI by default, I think that makes it more clear that the CLI is not required and not expected to be used, and that additional actions are required. I would say by default, tests shouldn't fail if the dependency isn't installed unless the user wants to explicitly enable this feature.

I would also argue that if we truly don't expect users to use the CLI, then we don't need all the capabilities that libargp provides. Short options that something like getopt provides seems reasonable from a functionality perspective.

Reply:
Actually, we could write around libargp. Probably we should.

I'm more concerned about depending on gcc specifically. 

Are we depending on gcc specifically as opposed to any other C compiler?

Reply:
I think we only depend on a C99 capable compiler. So users can decide whatever compiler they want that meets their needs.    That said, it might make sense to use clang in all our examples and github actions. It should work just the same and ASF would probably be more happy with that.

Reply:
We depend only on a C99 compliant compiler.  I've used clang as well as gcc on Ubuntu 20.04 and I can verify it works just the same.  I can update BUILD.md and the github actions to use clang although I'd have to find the correct steps and test they work on multiple platforms (e.g., install the clang-10 package and set CC=clang-10 and AR=llvm-ar-10 on Ubuntu 20.04).  Do you want those more complicated steps in the BUILD.md and the github actions now or as a TODO?  Likewise for replacing argp with getopt?  I'd prefer a TODO for the latter at least.

Reply:
Well, I would say these need to be fixed before any release, but you could do them in a separate JIRA ticket and separate PR.

Reply:
Agreed, that's my only hesitation with this PR. As long as it's fixed before the 3.1 release I'm a +1